rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,154 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+
+An IBLT has several parameters that define its characteristics. When comparing two IBLTs, these parameters MUST be the same.
+An overview of the parameters:
+
+* **buckets**: defines the size of the IBLT, much like the size of a bloom filter. The larger the size, the bigger the set difference can be. 
+* **Hk**: each key is hashed to select the buckets the key is inserted to. The hash function is applied on the previous output to create extra hash values. This continues until k different values are found. The hash function, its seed and the value for k MUST all be the same for the entire network.
+* **Hc**: each key is hashed to a checksum which is used to validate the inserted key during the deconstruction phase.
+* **data types**: the data types for *count*, *val_sum* and *hash_sum* MUST match ((un)signed, byte-size).
+
+The following parameters are used:
+
+* **buckets**: 1024
+* **Hc**: [murmur3](https://en.wikipedia.org/wiki/MurmurHash), 8 bytes, seed: 0x00
+* **Hk**: murmur3, 4 bytes, seed: 0x01, k = 6 
+* **key sum size**: 32 bytes
+* **bucket count field**: 4 bytes
+
+See Appendix A.1 for an explanation.
+
+### 4.1.3 IBLT page size
+
+Calculating an IBLT is relatively cheap, but it grows linearly with the number of transactions. This would harm performance of larger networks.
+The immutable nature of transactions make it possible to store intermediate IBLTs and use those as an optimization.
+The LC range parameters of the protocol should be adjusted accordingly to optimize use of these intermediate IBLTs.
+Therefore, they should not be chosen freely.
+
+Any LC value used in the protocol MUST be a multiple of 512 (minus 1, because the LC is 0 based). 
+The use of inclusive or exclusive range parameters must make sure that `end-start` is a multiple of 512.
+In the context of an IBLT this size will be referred to as the `IBLT page size` or `PSIZE` for short.
+
+### 4.2 Operation
+
+All messages and values mentioned in this chapter are scoped to a single connection between two nodes.
+*Alice* and *Bob* are used to represent two nodes connected to each other.
+
+The protocol generally operates as follows:
+
+1. Alice broadcasts at a set interval (§4.2.1):
+    * XOR of all known transaction references.
+    * Highest Lamport Clock value over all transactions (LC).
+    
+2. When receiving Alice's broadcast, Bob compares the XOR value from Alice with its own (§4.2.2):
+    * When the XOR is the same and the LC value equals BOB's highest Lamport Clock value: no action required.
+    * When different, send a message containing:
+      * LC value sent by Alice
+      * the IBLT that includes the Lamport Clock range of 0-LC
+      * Bob's highest Lamport Clock value
+
+3. When receiving the message, Alice subtracts Bob's IBLT from her own IBLT for the given range (§4.2.3):
+   * If not deconstructable, go to 1 and send values based on a lower Lamport Clock value.
```

Comment:
"a lower value" is vague for a spec. TODO?

Reply:
introduced the concept of a page in the spec (not in the message), so we can use the notion of a "previous page"